File: backend/app/api/v1/hashcat.py
from fastapi import APIRouter, HTTPException

from app.schemas.tools import HashcatHashModeResponse, HashcatTaskStatusResponse
from app.services.hashcat_service import hashcat_service
from app.services.tool_config_service import ToolDisabledError, tool_config_service

router = APIRouter()


@router.get("/status", response_model=HashcatTaskStatusResponse)
async def get_hashcat_status() -> HashcatTaskStatusResponse:
    try:
        tool_config_service.ensure_enabled("hashcat_gui")
    except ToolDisabledError as exc:
        raise HTTPException(status_code=403, detail=exc.message) from exc

    availability = tool_config_service.get_availability("hashcat_gui")
    return HashcatTaskStatusResponse(
        **hashcat_service.get_status(),
        enabled=availability.enabled,
        disabled_title=availability.disabled_title,
        disabled_message=availability.disabled_message,
    )


@router.post("/stop", response_model=HashcatTaskStatusResponse)
async def stop_hashcat_task() -> HashcatTaskStatusResponse:
    try:
        tool_config_service.ensure_enabled("hashcat_gui")
    except ToolDisabledError as exc:
        raise HTTPException(status_code=403, detail=exc.message) from exc

    availability = tool_config_service.get_availability("hashcat_gui")
    return HashcatTaskStatusResponse(
        **hashcat_service.stop_task(),
        enabled=availability.enabled,
        disabled_title=availability.disabled_title,
        disabled_message=availability.disabled_message,
    )


@router.get("/hash-modes", response_model=list[HashcatHashModeResponse])
async def list_hashcat_hash_modes() -> list[HashcatHashModeResponse]:
    try:
        tool_config_service.ensure_enabled("hashcat_gui")
    except ToolDisabledError as exc:
        raise HTTPException(status_code=403, detail=exc.message) from exc

    return [HashcatHashModeResponse(**item) for item in hashcat_service.get_hash_modes()]

File: frontend/src/api/hashcat.ts
import { api } from './logParser'
import type { HashcatHashMode, HashcatTaskStatus } from '../types/tools'

export async function getHashcatStatus(): Promise<HashcatTaskStatus> {
  const { data } = await api.get<HashcatTaskStatus>('/hashcat/status')
  return data
}

export async function stopHashcatTask(): Promise<HashcatTaskStatus> {
  const { data } = await api.post<HashcatTaskStatus>('/hashcat/stop')
  return data
}

export async function getHashcatHashModes(): Promise<HashcatHashMode[]> {
  const { data } = await api.get<HashcatHashMode[]>('/hashcat/hash-modes')
  return data
}
